addons/pos_loyalty/models/pos_session.py

```diff
@@ -113,3 +114,33 @@ def _get_pos_ui_product_product(self, params):
         self._process_pos_ui_product_product(products)
         result.extend(products)
         return result
+
+    def _pos_data_process(self, loaded_data):
+        super()._pos_data_process(loaded_data)
+        config = loaded_data['pos.config']
+        programs = loaded_data['loyalty.program']
```

Comment:
You need to take into account if there are loaded programs. Check _pos_ui_models_to_load, loyalty.program and related are only loaded if 'all_program_ids' return a set of records.